toast-tk-runtime/src/main/java/io/toast/tk/runtime/block/ArgumentBlockHelper.java

```diff
@@ -0,0 +1,116 @@
+
+
+public class ArgumentBlockHelper {
+
+	private static final Logger LOG = LogManager.getLogger(ArgumentBlockHelper.class);
+
+	public static boolean isArgumentFailureMessage(Exception error, Parameter[] params, Object[] inputArgList) {
+		if(!("argument type mismatch".equals(error.getMessage())) 
+				&& !(error instanceof InvocationTargetException && error.getCause() instanceof ClassCastException)) {
+			return false;
+		}
+		
+		String inputArg = "";
+		try {
+			for(Object input : inputArgList) {
+				inputArg = writeMessage(inputArg, input);
+			}
+		} catch (Exception e) {
+			LOG.error(e.getMessage(), e);
+		}
+		
+		String arg = "";
+		for(Parameter param : params) {
+			arg = writeMessage(arg, param.getParameterizedType());
+		}
+
+		String[] inputArgs = inputArg.split(System.lineSeparator());
+		String[] args = arg.split(System.lineSeparator());
+		
+		if(inputArgs.length != args.length) {
+			return true;
+		}
+		
+		for(int i = 0; i < inputArgs.length; i ++) {
+			String inputTemp = inputArgs[i];
+			String argTemp = args[i];
+			if(!inputTemp.equals(argTemp)) {
+				return true;
+			}
+		}
+		
+		return false;
+	}
+	
+	public static String getArgumentFailureMessage(Parameter[] params, 
+			Object[] inputArgList) {
+		String res = "Your inputs arguments are :" + System.lineSeparator();
+		try {
+			String inputArg = "";
+			for(Object input : inputArgList) {
+				inputArg = writeMessage(inputArg, input);
+			}
+			res += inputArg;
+		} catch (Exception e) {
+			LOG.error(e.getMessage(), e);
+		}
+		
+		res += System.lineSeparator() + "Instead of :" + System.lineSeparator();
+
+		String args = "";
+		for(Parameter param : params) {
+			args = writeMessage(args, param.getParameterizedType());
+		}
+		res += args;
+		
+		return res;
+	}
+	
+	@SuppressWarnings("rawtypes")
+	private static String writeMessage(String args, Object input) {
+		String className = input.getClass().getName().trim();
+		if(input instanceof List) {
+			if(((List) input).size() != 0) {
+				String name = ((List) input).get(0).getClass().getName().trim();
+				name = getClassName(name);
+				className = "List of " + name;
+			}
+		}
+		return writeMessage(args, className);
+	}
+	private static String writeMessage(String args, Type inputType) {
+		String className = inputType.getTypeName().trim();
+		if(className.startsWith("java.util.List")) {
+			className = className.substring("java.util.List<".length(), className.length() - ">".length());
+			className = "List of " + getClassName(className);
+		} else if(className.startsWith("java.util.Map")) {
+			className = className.substring("java.util.Map<".length(), className.length() - ">".length());
+			className = "Map of " + getMapClassName(className);
+		}
+		return writeMessage(args, className);
+	}
+	private static String writeMessage(String args, String className) {
+		String name = "- " + getClassName(className);
+		String res = args.equals("") ? name : args + System.lineSeparator() + name;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "res". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)